checkmk_extensions/agent/scheduler_runner.ps1

```diff
@@ -0,0 +1,63 @@
+# Enable StrictMode 3.0
+Set-StrictMode -Version 3.0
+
+class Config {
+    [bool]$rcc
+
+    Config() {
+        # Default constructor
+        # By default, we will not run in RCC
+        $this.rcc = $false
+    }
+
+    static [string] GetConfigContent($configFilePath, [ref]$errorCollection) {
+        try {
+            if (-Not (Test-Path -Path $configFilePath -PathType Leaf)) {
+                # Add the error message to the error collection
+                $errorCollection.Value.Add("Config file not found: $configFilePath")
+
+                # If we want to throw an ERRORRECORD object or a .NET exception, we can use the something like:
+                # $errorCollection.Value.Add((New-Object System.IO.FileNotFoundException))
+                return $null
+            }
+
+            $configFileContent = Get-Content -Raw -Path $configFilePath
```

Comment:
Will `Get-Content` throw an error if the file is missing or if the read itself fails?

Reply:
Ok, so then we don't have to check ourselves if the file exists, right? Get-Content will tell us if it doesn't.

Reply:
Yes, we can make sure that it raises it. I will just add a -ErrorAction flag to it.